messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -262,4 +263,44 @@ public MessageCheckpoint getMessageCheckpoint() {
   public MessageCheckpoint buildStartPartitionCheckpoints() {
     return null;
   }
+
+  @Override
+  public synchronized Long getReaderBackLog() throws IOException {
+    Long collectorStreamPendingSize =0l,localStreamPendingSize = 0l;
+    int retryCount = 0;
+    int maxRetryThreshold = 10;
+    while (cReader.getCurrentFile() == null && retryCount < maxRetryThreshold){
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        LOG.info("Sleep Interrupted while waiting for reader initialisation- Ignoring");
+      }
+      retryCount++;
+    }
+    //get collector reader remaining size
+    if (cReader.getCurrentFile() != null) {
+      collectorStreamPendingSize += cReader.getPendingSize(cReader.getCurrentFile());
+      LOG.info("Pending Size inside collector output location starting from collector file: " +
+              cReader.getCurrentFile() + " is "+ collectorStreamPendingSize);
+    }
+    //get local reader remaining size
+    retryCount = 0;
+    while (lReader.getCurrentFile() == null && retryCount < maxRetryThreshold){
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        LOG.info("Sleep Interrupted while waiting for reader initialisation- Ignoring");
```

Comment:
Sleep interrupted while waiting for local stream reader initialization. Ignoring the interrupt.